splash/resources.py

```diff
@@ -38,7 +39,7 @@ def render(self, request):
             return Resource.render(self, request)
         except BadOption as e:
             request.setResponseCode(400)
-            return str(e) + "\n"
+            return str(e).encode('utf-8') + b"\n"
```

Comment:
`str(e).encode('utf8')` is not correct in Python 2: you're calling .encode method for bytestring, and in thiscase Python first tries to convert it to unicode using sys.getdefaultencoding (usually ascii), and only then encode to the specified encoding. So it can fail with `UnicodeDecodeError` if str(e) contains non-ascii data.